src/Apps/Order/Routes/Shipping2/Components/AddressModal2.tsx

```diff
@@ -330,37 +429,45 @@ export const AddressModal: React.FC<AddressModalProps> = ({
   )
 }
 
-export enum AddressModalActionType {
-  EDIT_USER_ADDRESS = "editUserAddress",
-  CREATE_USER_ADDRESS = "createUserAddress",
-}
-
-export type AddressModalAction =
-  | {
-      type: AddressModalActionType.CREATE_USER_ADDRESS
-    }
-  | {
-      type: AddressModalActionType.EDIT_USER_ADDRESS
-      address: SavedAddressType
-    }
-
-const MODAL_TITLE_MAP: Record<AddressModalActionType, string> = {
-  createUserAddress: "Add address",
-  editUserAddress: "Edit address",
-}
-
+// two different error messages for the same error?
 const SERVER_ERROR_MAP: Record<string, Record<string, string>> = {
   "Validation failed for phone: not a valid phone number": {
-    field: "phoneNumber",
+    field: "attributes.phoneNumber",
     message: "Please enter a valid phone number",
   },
   "Validation failed: Phone not a valid phone number": {
-    field: "phoneNumber",
+    field: "attributes.phoneNumber",
     message: "Please enter a valid phone number",
   },
 }
 
+const validationSchema = Yup.object().shape({
+  attributes: Yup.object().shape(ADDRESS_VALIDATION_SHAPE),
+})
+
 export const GENERIC_FAIL_MESSAGE =
   "Sorry, there has been an issue saving your address. Please try again."
 
-const validationSchema = Yup.object().shape(ADDRESS_VALIDATION_SHAPE)
+const handleGravityErrors = (
+  errors: ReadonlyArray<{ message: string }>,
```

Comment:
Gravity mutation helpers in `useUserAddressUpdates` now always return a type something like `{data: T} | {errors: [{message?: string}]} | null` (null if no action was performed). This lets us handle the errors a bit more consistently.